app/src/main/java/cs5150athletetracking/com/athletetracking/LocationRecorder/LocationRecorder.java

```diff
@@ -114,12 +131,45 @@ public boolean hasError(){
         return error.get() != LocationRecorderError.NONE;
     }
 
+    // ALL UI code belongs on the main thread, hence this ugliness
```

Comment:
These are pretty ugly, but Android will crash or otherwise mess up if you don't run all UI code on the main/UI thread